text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,765 @@
+is largely borrowed from
+[the initial proposal](https://docs.google.com/document/d/1VUdBRInLEhO_0ABAoiLEssB1CQO_IcD5zDnaMEha42w/edit)
+that was accepted for entities SIG formation.
+
+This OTEP is step 1 in introducing the entities data model. Follow up OTEPs will add
+further data model definitions, including the linking of Resource information
+to entities.
+
+## Design Principles
+
+- Consistency with the rest of OpenTelemetry is important. We heavily favor solutions
+  that look and feel like other OpenTelemetry data models.
+
+- Meaningful (especially human-readable) IDs are more valuable than random-generated IDs.
+  Long-lived IDs that survive state changes (e.g. entity restarts) are more valuable than
+  short-lived, ephemeral IDs.
+  See [the need for navigation](https://docs.google.com/document/d/1Xd1JP7eNhRpdz1RIBLeA1_4UYPRJaouloAYqldCeNSc/edit#heading=h.fut2c2pec5wa).
+
+- We cannot make an assumption that the entirety of information that is necessary for
+  global identification of an entity is available at once, in one place. This knowledge
+  may be distributed across multiple participants and needs to be combined to form a
+  globally unique identifier.
+
+- Semantic conventions must bring as much order as possible to telemetry, however they
+  cannot be too rigid and prevent real-world use cases.
+
+## Data Model
+
+We propose a new concept of Entity.
+
+Entity represents an object of interest associated with produced telemetry:
+traces, metrics or logs.
+
+The Entity that produces the telemetry is called the Producing Entity. For example
+telemetry produced using OpenTelemetry SDK is normally associated with a Service.
+The Service is the Producing Entity. Similarly, OpenTelemetry defines system metrics
+for a host. The Host is the Producing Entity in this case.
+
+Entities may be also associated with produced telemetry indirectly, typically by
+virtue of being related to the Producing Entity. For example a Service that produces
+telemetry is also related with a process in which the Service runs, so we say that
+the Service entity is related to the Process entity. The process normally also runs
+on a host, so we say that the Process entity is related to the Host entity.
+
+Note: subsequent OTEPs will define how the Producing Entity is associated with
+traces, metrics and logs and how relations between entities will be specified.
+See [Future Work](#future-work).
+
+The data model below defines a logical model for an entity (irrespective of the physical
+format and encoding of how entity data is recorded).
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>string
+    </td>
+    <td>Defines the type of the Entity. MUST not change during the
+lifetime of the entity. For example: "service" or "host". This field is
+required and MUST not be empty for valid entities.
```

Comment:
Yes, we should look into that as part of that Open Question. My current thinking is that it is not a runtime data model question per se, but rather a matter of modelling the entity semantic conventions (aka our type definitions). I may change my opinion if we show use cases where having this information at runtime is useful because we have entities the type of which within the hierarchy may change dynamically and cannot be a static knowledge recorded in semantic conventions. So far I can't see a use case like that.